ui/src/main/kotlin/edu/artic/ui/BaseFragment.kt

```diff
@@ -37,21 +44,38 @@ abstract class BaseFragment : Fragment() {
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         return inflater.inflate(layoutResId, container, false)
-
     }
 
+    protected open fun hasTransparentStatusBar(): Boolean = false
+
+    protected open fun hasHomeAsUpEnabled(): Boolean = true
+
     private fun updateToolbar(view: View) {
         toolbar = view.findViewById(R.id.toolbar)
         if (toolbar != null) {
             baseActivity.setSupportActionBar(toolbar)
             baseActivity.supportActionBar?.apply {
                 setDisplayShowTitleEnabled(true)
-                setDisplayHomeAsUpEnabled(true)
+                setDisplayHomeAsUpEnabled(hasHomeAsUpEnabled())
                 setDisplayShowHomeEnabled(true)
             }
 
             baseActivity.title = title
         }
+
+        collapsingToolbar = view.findViewById(R.id.collapsingToolbar)
+        collapsingToolbar?.let {
+            val toolbarTextTypeFace = ResourcesCompat.getFont(requireContext(), R.font.ideal_sans_medium)
+            it.setCollapsedTitleTypeface(toolbarTextTypeFace)
+            it.setExpandedTitleTypeface(toolbarTextTypeFace)
+        }
+
+        if (hasTransparentStatusBar()) {
+            activity?.setWindowFlag(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS, false)
```

Comment:
`requireActivity()`

Reply:
for all `activity?`

Reply:
Yes, it updates the activity's status bar. It will update back to primaryColorDark when necessary.
Is there another way to do this?